### PR TITLE
chore: Update standards data

### PR DESCRIPTION
**Pull request by ermshiperete**
Also add source of files to readme.

@keymanapp-test-bot skip

### CODE REVIEW

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Nearly LGTM -- however I think we should be explicit about which versions of data we are using. Can you make sure you add that to the README for each file and then add the same data to the release notes?

**Comment by mcdurdin** [comment]
Also, failing builds for Windows and Developer:

```
bin\Win32\Release\build_standards_data.exe langtags 

C:\BuildAgent\work\7ac43416c45637e9\keyman\resources\standards-data\langtags\langtags.json C:\BuildAgent\work\7ac43416c45637e9\keyman\common\windows\delphi\standards\Keyman.System.Standards.LangTagsRegistry.pas
C:\BuildAgent\work\7ac43416c45637e9\keyman\common\windows\delphi\standards\Keyman.System.Standards.LangTagsRegistry.pas
EAccessViolation: Access violation at address 005164BE in module 'build_standards_data.exe'. Read of address 00000000
```

And Linux: failing test_canonical_language_code_utils.CanonicalLanguageCodeUtilsTests.test_FindBestTag_ValidCases

File: common/windows/delphi/Makefile [hunk 1]
```diff
@@ -4,6 +4,6 @@ global-versions:
 # Build our root version.txt; TODO: replace with build.sh deps
     cd $(KEYMAN_ROOT)\common\windows
     $(MKVER_U) $(KEYMAN_ROOT)\common\windows\delphi\general\keymanversion_build.in $(KEYMAN_ROOT)\common\windows\delphi\general\keymanversion_build.inc
-    $(GIT_BASH_FOR_KEYMAN) $(KEYMAN_ROOT)/common/include/build.sh clean configure build
+    $(GIT_BASH_FOR_KEYMAN) $(KEYMAN_ROOT)\common\include\build.sh clean configure build
```
Comment (mcdurdin) [inline common/windows/delphi/Makefile:7]: This change is unnecessary -- forward slashes are appropriate.

Comment (ermshiperete) [reply common/windows/delphi/Makefile:7]: That's what I thought as well, but forward slashes things didn't work when I tried from the command line. The problem is that `cmd.exe` doesn't like paths with forward slashes.

**Comment by ermshiperete** [comment]
> > Nearly LGTM -- however I think we should be explicit about which versions of data we are using. Can you make sure you add that to the README for each file and then add the same data to the release notes?
> 
> Would it be preferable to consolidate the standards-data readme's to one file to maintain? (changelog in a separate file)

@mcdurdin?

File: common/windows/delphi/Makefile (hunk 1 above)
Comment (mcdurdin) [reply common/windows/delphi/Makefile:7]: OK, I can see that if the `GIT_BASH_FOR_KEYMAN` variable is not set we'll run into the problem you describe.

We should be requiring it for Windows builds, although it is currently documented as optional. We haven't really tested not having it, and I think we'll run into other problems such as non-linear builds (e.g. we need `start /wait` on this command without the variable defined; it also pops an awful bash window for each command). So it makes the builds much less reliable if it isn't present. 

So have you set your `GIT_BASH_FOR_KEYMAN` environment variable? See docs/build/windows.md for details.

I'd prefer not to be changing makefiles in this PR.

Comment (ermshiperete) [reply common/windows/delphi/Makefile:7]: Ah, ok. With `GIT_BASH_FOR_KEYMAN` things work, so I'll revert the changes to `Makefile`.

File: resources/standards-data/readme.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+| langtags-subtag-registry   | 2023-05-11  |
+| ldml-keyboards             | techpreview |
+| unicode-character-database | 15          |
+| windows-lcid-to-bcp-47     | ?           |
+
+## File Sources
+
+### Ethnologue
+
+??? (I'm not sure how to update this file or where it comes from)
```
Comment (mcdurdin) [inline resources/standards-data/readme.md:22]:
```suggestion
This data comes from https://www.ethnologue.com/#resources, LanguageIndex.tab. It has been hand-converted from TSV to CSV, and the header line has been replaced.

```

File: resources/standards-data/readme.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# standards-data
+
+The subfolders of this directory contain data from various standards
+that are used in Keyman.
```
Comment (mcdurdin) [inline resources/standards-data/readme.md:4]:
```suggestion
The subfolders of this directory contain data from various standards
that are used in Keyman. If you update any files, be sure to update
the version data in this file also.
```

File: resources/standards-data/readme.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## File Sources
+
+### Ethnologue
+
+??? (I'm not sure how to update this file or where it comes from)
+
+### iso639-3
+
+The file can be update with:
```
Comment (mcdurdin) [inline resources/standards-data/readme.md:26]:
```suggestion
The file can be updated with:
```

File: resources/standards-data/readme.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+### langtags
+
+The file is retrieved from <https://ldml.api.sil.org/langtags.json>:
+
+```bash
+wget https://ldml.api.sil.org/langtags.json -O langtags/langtags.json
+```
+
```
Comment (mcdurdin) [inline resources/standards-data/readme.md:39]:
```suggestion

See also [langtags/readme.md](langtags/readme.md).

```

File: resources/standards-data/readme.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+```bash
+wget https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry \
+  -O language-subtag-registry/language-subtag-registry
+```
+
+### ldml-keyboards
+
+This data is from <https://github.com/unicode-org/cldr.git>
+
```
Comment (mcdurdin) [inline resources/standards-data/readme.md:52]:
```suggestion

See also [ldml-keyboards/readme.md](ldml-keyboards/readme.md).

```

**Review by mcdurdin — APPROVED** [review]
LGTM

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.138-alpha](https://keyman.com/downloads/releases/alpha/17.0.138)

**Pull request merged**